gnovm/cmd/gno/transpile.go

```
@@ -169,17 +195,39 @@ func execTranspile(cfg *transpileCfg, args []string, io commands.IO) error {
 	return nil
 }
 
-func transpilePkg(pkgPath importPath, opts *transpileOptions) error {
-	if opts.isTranspiled(pkgPath) {
+// transpilePkg transpiles all non-test files at the given location.
+// Additionally, it checks the gno.mod in said location, and skips it if it is
+// a draft module
+func transpilePkg(dirPath string, opts *transpileOptions) error {
+	if opts.isTranspiled(dirPath) {
+		return nil
+	}
+	opts.markAsTranspiled(dirPath)
+
+	gmod, err := gnomod.ParseAt(dirPath)
+	if err != nil && !errors.Is(err, gnomod.ErrGnoModNotFound) {
+		return err
+	}
+	if err == nil && gmod.Draft {
+		if opts.cfg.verbose {
+			fmt.Fprintf(os.Stderr, "%s (skipped, gno.mod marks module as draft)\n", filepath.Clean(dirPath))
+		}
+		opts.skipped = append(opts.skipped, dirPath)
 		return nil
 	}
-	opts.markAsTranspiled(pkgPath)
 
-	files, err := filepath.Glob(filepath.Join(string(pkgPath), "*.gno"))
+	// XXX(morgan): Currently avoiding test files as they contain imports like "fmt".
+	// The transpiler doesn't currently support "test stdlibs", and even if it
+	// did all packages like "fmt" would have to exist as standard libraries to work.
+	// Easier to skip for now.
+	files, err := listNonTestFiles(dirPath)
 	if err != nil {
 		log.Fatal(err)
```

shall we return the error instead?